stdlib/seq.mc

```diff
@@ -1,8 +1,18 @@
 include "option.mc"
 
-let head = lam s. nth s 0
-let tail = lam s. slice s 1 (length s)
 let null = lam seq. eqi 0 (length seq)
+let head = lam seq. get (splitAt seq 1).0 0
+let tail = lam seq. (splitAt seq 1).1
+let last = lam seq. get (splitAt seq (subi (length seq) 1)).1 0
+let init = lam seq. (splitAt seq (subi (length seq) 1)).0
```

Comment:
Is `splitAt` clever enough to do the efficient thing here?

Reply:
I believe so, see the implementation of `split_at` in `mseq.ml`